### ISSUE TITLE
Thoughts on the bot

### ISSUE THREAD

**Issue by zvei** [prinzstani/CookieBot#9]
This is more of a insight then a issue but I've been thinking about the bot. I feel that due to the additional bake xx cookies in one ascension. I think getting all shadow achievements and cheating in sugar lumps after getting the bake 1 duodecillion cookies in 1 ascension and have at least 350 of every building achievement is a better option. My reasoning behind this is at this point everything is a big grind and even more so if you are aiming for 1 quindecillion cookies baked in one ascension which is currently impossible with legit play. So you would need all the sugar lumps you could get at that point to boost CPS. Unfortunately, I understand this would require a lot of time and coding and you may not have the time and along with the garden update it's a big ask for you.

**Comment by prinzstani** [comment]
I am working on 2.01 and will keep this in mind.

**Comment by zvei** [comment]
How's the progress coming along? :)

**Comment by prinzstani** [comment]
It is possible that I have a partial new version of the bot including the unlocking of the plants in about one weeks time. Still trying to see whether I should keep the sugar lump cheating or not. It seems it is possible to get all sugar lumps in a reasonable time now. It seems that the problem still exists for the golden sugar lump, so here cheating might be needed.
And then of course there is some problem to get all the grinding achievements in a reasonable time.

**Comment by zvei** [comment]
I believe Orteil will do something about the grinding achievements so you won't have to worry about that much. Plus gardens has a ton of content therefore you should just focus on gardens. If you get the biggest obstacle out of the way then everything should work out

**Comment by zvei** [comment]
Can you explain what the bot is doing please? It says ascending 1000 times but it's not ascending. Is it aiming for something else perhaps that hasn't been shown.

Mi4wMTA2fHwxNTI0NTk3NjQ0OTE0OzE1MjI0NDUxODQwOTI7MTUyNTI2Nzg0NDg5MTtBdXRvbWF0ZWQgQ2FuYWRpYW5Gcmllczt1eWZ2dnwxMDExMTExMTEwMDEwMDExMHwxLjU2ODg4MzUzODA3ODExOGUrMzc7Ny42MjEzMDc4MDk4NTcyNzdlKzM5Ozc5NTQ1NzsxMTU4NTsxLjE0NTk1MjIwNzE3NTgwODRlKzM5OzE2Mjs0OzE4OzEuMTk0MDkyMDY0MzczMjY0NmUrMzk7MTswOzA7MDstMTsxMTI7MjMyODszLjQ5OTQ1NjI1ODkxMDY2M2UrMzg7MjYwOzE0OzIyMjg7MTA2MjY2NDs5O2NocmlzdG1hczszLjAyMDQ5NDAzMDQxMzk2OTdlKzM2OzEyOzEwNjA5MTE3Nzc7MjkxODAyMjE7MTAzMTczMTU1NjswOzA7NDAwOzQ0Mjs0NDE7MjI2OzcyOzIyOzE1OzE7MDswOzA7MDswOzMyOzE1MjUyNjMwMjA5NDI7MDswO3w1ODIsOTMyLDQuMzk5MDk2NTU0MDA1OTMxZSszNSwwOzU1NSw5MDUsOS40MDE1NDMyMDEwMjQ5MzdlKzMzLDA7NTMwLDg4MCw3LjgwNTc2NjUwODUyOTY2N2UrMzQsMSwxNTI1MjY3ODUzNTYwOjA6MTUyNDU5NzY0NTQyNTowOjA6MDowOjA6IDEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAgMDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6OzU1MCw5MDAsNC44NDAwOTI4MTIwODc0OTFlKzM2LDA7NTE1LDg2NSw5Ljg1NzQ3NDExNjA4ODQxMWUrMzUsMDs0ODgsODM4LDIuNTM5NTIxNDI0ODAzNTUxZSszNSwwOzQ4MCw4MzAsMS4xMzQ5Njk3NTcyMDU1OTI3ZSszNiwxLDgvMS83LzcgMiAxNTI1MjY1MTE0OTc4IDA7NDUzLDgwMyw0LjM5NzkxODk4MDM4NTY0NGUrMzUsNyw3My40NTE2MTMyMzUyOTM3IDE3MCA3NjkgMDs0MjcsNzc3LDEuNjI2Mzg2NTI4NDA1MjI5ZSszNSwwOzQyMCw3NzAsOC40Mzg2NjM2MTEwNDY4MjZlKzM1LDA7NDE0LDc2NCw0LjczMzE0NDA4Njc1OTk5OWUrMzYsMDs0MDcsNzU3LDIuODYzODE1NjcwOTgzMzU0ZSszNywwOzQwMCw3NTEsMi40NzQ5NjUxMjU4NjI3MTA1ZSszOCwwOzM3OCw3MjgsNy4zNTUwMDk4MzI0MTQ0MzI1ZSszNywwOzM2Niw3MTgsMi41NjMzMDIyNzc3NDE3OTg0ZSszOCwwO3wxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAxMTAwMDAxMTExMTExMTAwMTExMTExMDAwMDAwMTEwMDExMTExMTAwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDEwMTAwMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMTAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTExMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDAwMDExMTExMTExMTExMTExMTExMTExMDAxMTExMDAxMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMTAxMTExMTExMTExMTEwMDExMTExMTExMTExMTExMTExMTExMDAwMDAwMDAwMDAwMDAxMTAwMDB8MTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDAwMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAxMTAwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMDAwMDAwMDAwMDAwMTEwMTExMTExMTExMTExMDExMTExMTExMTExMTExMTAwMTExMTExMDAwMDAwMDAxMTAwMDAwMDExMDAwMDAxMTAwMDAwMDAxMDB8%21END%21

**Comment by prinzstani** [comment]
It waits for Sucralosia Inutilis to be purchased. he reason for this is that then a lot of prestige levels are available. As it stands now, the bot will wait until it has run for 10 days, and then it will ascend, thereby starting the 1000 times by buying Sucralosia. In fact, maybe it takes even a bit longer as you need 2.6 billion prestige for Sucralosia. Just wait, everything will be fine.
I think this will be easier to see when the status of the bot is displayed better.
Currently this is not top priority, as the gardens are more urgent.
If you do not want to wait, you can trigger ascend which might help for byuing Sucralosia.

**Comment by zvei** [comment]
Thanks

**Comment by zvei** [comment]
What does the bot do when idling?

**Comment by prinzstani** [comment]
Idling means the bot is done with the work and just produces more cookies. This will probably not happen until the garden is implemented.
At night, the bot does nothing.

**Comment by SearchAndDestroy** [comment]
I am using the bot (beta version) since a few days and I agree that a better way to see what it's doing/waiting on would be nice, even just messages in the console.
I also think that you should focus on the garden.

**Comment by Buti1986** [comment]
Since I lost my old cookie clicker save, I think about to give a try to this bot just for fun.
I just start (have the neverclick-harcore archieves already) but I have a couple of questions
I have 2 sugar lumps ready to use, should not the bot use them to get more CPS or just unlock the garden? 
How does the bot calcule what buy next? It looks like it only buy upgrades if they are cheaper than the next build it wants to buy, I know eventually this is enough to get any goal, but can't you get a bit more efficience buy plan?

EDIT: Also, is already switching the night mode off possible or is still a WIP? If so, how can I do that? I don't see any options anywhere

**Comment by prinzstani** [comment]
The sugar lump problem is solved in the current beta.
Switching off night mode will be part of the options-issue, which is currently not even planned. Make a comment there, and it will get higher priority.
Proper handling of upgrades is quite complex, and not something planned for the near future. I am not sure I will work on that.

**Comment by zvei** [comment]
So after the gardens update,do you have any future plans for the bot?

**Comment by prinzstani** [comment]
I just implemented the current status information of the bot. Still ongoing tests of the garden release against a real playthrough. Will maybe work on cheating somewhat when the grinding is too serious ...

**Comment by zvei** [comment]
The grinding will help towards the 1 year legacy achievement

**Comment by prinzstani** [comment]
New cheating of golden cookies is introduced which reduces the time for grinding.
This way the shadow achievements still come at the end.
The one-tear achievement is considered outside the work of the bot.

**Comment by zvei** [comment]
Do you have any extra features you plan on implementing in the bot?
Will you be working on overall efficiency for all aspects or is that not a priority?

**Comment by prinzstani** [comment]
Just bugfixing and small improvements now - there is announced a big release with the dungeons - this will take some more work.

**Comment by zvei** [comment]
Good luck on the dungeons :)
